### DISCUSSION TITLE
How to use SwiftUI .scrollPosition?

### DISCUSSION THREAD

**Discussion by andreivladmatei** [gonzalezreal/swift-markdown-ui#371] [Q&A] [answered]
Hello!

Is there a way to use the ScrollView's scrollPosition modifier to jump to a certain paragraph (or element)?

Let's say I have a string composed from 10 paragraphs in a scrollview, rendered by MarkdownUI and I want to scroll directly to paragraph 4 (for which I can determine start character index).

**Comment by gonzalezreal** [comment]
I am not sure if this is what you are looking for, but the Demo project shows how to implement heading anchors (ie. links that scroll to a specific heading in a document).

https://github.com/gonzalezreal/swift-markdown-ui/blob/a9c7615fb50323069c2979c69263973aa1b24a8f/Examples/Demo/Demo/RepositoryReadmeView.swift#L126

**Reply by andreivladmatei** [reply]
thank you, using your example I managed to create a button that programmatically scrolls to a heading. (like in your example, but without links, only with `proxy.scrollTo("configuring-the-command-line-tool", anchor: .top)`

What I really need is to scroll to the beginning of various blocks (e.g. a paragraph). I can't manage to do that, what am I missing?

**Reply by gonzalezreal** [reply, marked as answer]
Blocks like headings or paragraphs are rendered to SwiftUI views. The `ScrollViewProxy` scrolls to the first view identified by the given identifier.

Headings are automatically identified by their text:
https://github.com/gonzalezreal/swift-markdown-ui/blob/a9c7615fb50323069c2979c69263973aa1b24a8f/Sources/MarkdownUI/Views/Blocks/HeadingView.swift#L21

I haven't tested this, but it should be possible to add an identifier to each paragraph by overriding the paragraph style in the current theme:

```swift
Markdown("...")
   .markdownBlockStyle(\.paragraph) { configuration in
      configuration.label
          .id(/* find some meaningful id here based on configuration.content */)
   }
```

**Reply by andreivladmatei** [reply]
This was what I was looking for. Thank you 😄
